File: frontend/src/types/schemaUmgebung.ts
export interface SchemaUmgebung {
  id: string
  name: string
  system: boolean
  beschreibung: string | null
  aktiv: boolean
  createdAt: string
  updatedAt: string
}

export interface SchemaUmgebungCreateFormData {
  name: string
  beschreibung?: string
}

export interface SchemaUmgebungUpdateFormData {
  name: string
  beschreibung?: string
  aktiv: boolean
}

File: backend/edugate-core/src/main/java/de/svws_nrw/edugate/core/domain/SchemaUmgebung.java
package de.svws_nrw.edugate.core.domain;

import java.time.Instant;
import java.util.UUID;

/**
 * Eine vom Betreiber gepflegte Schema-Umgebung (ADR-012: "Umgebung" ist eine erweiterbare
 * Betreiber-Konvention, kein geschlossenes Enum). Mandantenübergreifende Betriebsressource, kein
 * Mandanten-Objekt - analog zu {@link SvwsInstanz} (ADR-011). {@code system}-Einträge (z. B.
 * {@link SchemaNamingKonstanten#PRODUKTIV}) sind reserviert: Name und Aktiv-Status sind
 * unveränderlich.
 */
public record SchemaUmgebung(
    UUID id,
    String name,
    boolean system,
    String beschreibung,
    boolean aktiv,
    Instant createdAt,
    Instant updatedAt
) {
}
